### PR TITLE
GHA build and push docker

### PR DESCRIPTION
**Pull request by kgaonkar6**
#### What scientific question is your analysis addressing?
Pushes the updated docker image built with Dockerfile in `dev` to kfdrc/open-pedcan:latest

### CODE REVIEW

File: .github/workflows/build-and-push-docker.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      -
+        name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: kfdrc/open-pedcan:latest
```
Comment (dmiller15) [inline .github/workflows/build-and-push-docker.yml:31]: As long as you are ok losing old versions of the docker image this is fine.

File: .github/workflows/build-and-push-docker.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: Docker Build and Push
+
+# Controls when the action will run. 
+# Triggers the workflow for a push to dev
+# with changes in Dockerfile.
+on:
+  push:
```
Comment (dmiller15) [inline .github/workflows/build-and-push-docker.yml]: Does GHA have an `on merge`?

**Review by dmiller15 — COMMENTED** [review]
Some comments.

**Comment by dmiller15** [comment]
One thing of note. You aren't doing checkout. My understanding is that you need to checkout the repo so you can have it in the GHA env.

**Comment by dmiller15** [comment]
How long is a typical docker build? Is there a reason to be building the docker every time? Wouldn't you just want to do that when you make changes to the dockerfile?

File: .github/workflows/build-and-push-docker.yml (hunk 1 above)
Comment (kgaonkar6) [reply .github/workflows/build-and-push-docker.yml:31]: I think we are ok with losing the old version for this docker image

**Comment by kgaonkar6** [comment]
Because I have `paths: [ Dockerfile ]` in the `on` event section,  the action will only run in here and the previous PR only if the Dockerfile changes.  I've tested this in my local repo , I wonder if there are ways a reviewer can test these GHA updates .. checking 

Update: didn't find any quick ways to review github actions

File: .github/workflows/build-and-push-docker.yml (hunk 2 above)
Comment (kgaonkar6) [reply .github/workflows/build-and-push-docker.yml]: checking

Comment (kgaonkar6) [reply .github/workflows/build-and-push-docker.yml]: I guess this is how we would setup the action on merge  as per this blog post https://brennerm.github.io/posts/trigger-github-actions-on-pr-close.html

implemented the above in this commit https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/14/commits/55e5728c6fd11a17b5ff0df7486e5b6575bed9f2

Comment (dmiller15) [reply .github/workflows/build-and-push-docker.yml]: Interesting. Good to know.

**Pull request merged**